### DISCUSSION TITLE
Switching from docker to docker rootless while keeping the existing containers and data

### DISCUSSION THREAD

**Discussion by tomassplatch** [nextcloud/all-in-one#3223] [Questions] [answered]
Hello, I have been running a single AIO instance on my Ubuntu Server 22.04. I need to add a supplementary instance https://github.com/nextcloud/all-in-one/blob/main/multiple-instances.md) with a different domain to my setup and would like to use the docker rootless option. I use Caddy for reverse proxy setup (AIO on subdomain, Wordpress website on the main domain). Sorry I am not too experienced, but I wanted to ask what would be the proper way to do it. If I follow the official instructions and install a rootless docker for all users, will I be able to keep the existing data of the existing instance (I use a custom NEXTCLOUD_DATADIR in the home directory of the user)? 

Or would should I rather reset the old instance and restore it from backup?

To resume, here is the current state:
Instance1: user1 (member of the root group): NEXTCLOUD_DATADIR in /home/user1, APACHE_PORT=11000, cloud.domain1.com

Desired state:
Instance1: user1 (member of the root group), NEXTCLOUD_DATADIR in /home/user1, APACHE_PORT=11000, cloud.domain1.com - keeping the existing data
Instance2: user2, NEXTCLOUD_DATADIR in /home/user2, APACHE_PORT=12000, cloud.domain2.com

I hope it is clear, I would be very grateful for any recommendations or potential pitfalls I should be aware of...

**Comment by szaimen** [comment, marked as answer]
> Or would should I rather reset the old instance and restore it from backup?

I would recommend this
